core/src/components/header/header.tsx

```diff
@@ -85,7 +85,7 @@ export class Header implements ComponentInterface {
 
     if (hasCondense) {
       const pageEl = this.el.closest('ion-app,ion-page,.ion-page,page-inner');
-      const contentEl = pageEl ? findIonContent(pageEl) : null;
+      const contentEl = pageEl ? pageEl.querySelector('ion-content') : null;
```

Comment:
Why was this change made? Isn't this reverting the custom scroll target behavior for `condense`?

Reply:
This change was made to have the intersection observer attach the root container as the `ion-content` instead of the virtual scroll target.

e.g.: https://github.com/ionic-team/ionic-framework/blob/23359198231f8b205f2d8578191252f3a411554b/core/src/components/header/header.tsx#L145-L189

The condense feature works by subscribing to different thresholds when the toolbar intersects the root element. 

We are still querying the custom scroll target's scroll container in the implementation: https://github.com/ionic-team/ionic-framework/blob/23359198231f8b205f2d8578191252f3a411554b/core/src/components/header/header.tsx#L154

This is different than the fade implementation, because the fade feature only requires listening to the `scroll` event from our scroll container. 

Prior to this change, if `ion-header` is outside of the scroll container, e.g.:
```html
<ion-content>
  <ion-header></ion-header>
  <div class="ion-content-scroll-host"></div>
</ion-content>
```

The intersection observer would attach to the `.ion-content-scroll-host`. `ion-header` would never intersect this container, since it sits outside of it. This made the condense feature never work with our expected implementation pattern.